### ISSUE TITLE
Ignore links inside `<script>` tags

### ISSUE THREAD

**Issue by WhyNotHugo** [untitaker/hyperlink#148]
The script in here should be ignored; it's not _actual_ HTML.

```htmldjango
<script id="algolia__template" type="text/template">
{% raw %}
  <div class="algolia__result">
    <a class="algolia__result-link" href="{{ full_url }}#algolia:{{ css_selector }}">{{{ _highlightResult.title.value }}}</a>
    {{#posted_at}}
    <div class="algolia__result-date">{{ posted_at_readable }}</div>
    {{/posted_at}}
    <div class="algolia__result-text">{{{ _highlightResult.text.value }}}</div>
  </div>
{% endraw %}
</script>
```

**Comment by untitaker** [comment]
Yeah unfortunately this is a known issue. It's a lack of features in the underlying html parsing library, specifically absence of a proper tree builder

**Comment by untitaker** [comment]
underlying html5gum issue: https://github.com/untitaker/html5gum/issues/11

**Comment by untitaker** [comment]
Should be fixed in 0.1.22. There is still no proper tree builder in html5gum, but the library now includes workarounds that work in the majority of cases. Certainly good enough for hyperlink.

**Comment by untitaker** [comment]
There was a bug potentially affecting this, and causing link tags in `<script>` **still** to be read. It was fixed with #181, which will appear in 0.1.41
